File: java-back-server/src/main/java/space/borisgk/taxi/api/service/MessageService.java
package space.borisgk.taxi.api.service;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Service;
import space.borisgk.taxi.api.model.entity.Message;

import javax.persistence.EntityManager;
import javax.persistence.criteria.CriteriaBuilder;

@Service
public class MessageService extends AbstractCrudService<Message> {
    public MessageService(JpaRepository<Message, Integer> repository, EntityManager em, CriteriaBuilder cb) {
        super(repository, em, cb);
    }
}

File: kotlin-back-server/src/main/kotlin/space/borisgk/taxi/api/service/MessageService.kt
package space.borisgk.taxi.api.service

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import space.borisgk.taxi.api.model.entity.Message
import space.borisgk.taxi.api.repository.MessageRepository

@Service
class MessageService {

    @Autowired
    private val messageRepository: MessageRepository? = null

    fun saveMessage(message: Message): Message {
        return messageRepository!!.saveAndFlush(message)
    }

    fun getAllByTripId(tripId: Int?): List<Message> {
        return messageRepository!!.getAllByTripId(tripId)
    }
}
